src/server/GraphQLServer.ts

```diff
@@ -1,18 +1,419 @@
+            this.collectErrorMetricsFunction = options.collectErrorMetricsFunction || this.defaultCollectErrorMetrics
+            this.schemaValidationFunction = options.schemaValidationFunction || validateSchema
+            this.parseFunction = options.parseFunction || parse
+            this.validationRules = options.validationRules
+            this.validationTypeInfo = options.validationTypeInfo
+            this.validationOptions = options.validationOptions
+            this.removeValidationRecommendations = options.removeValidationRecommendations === undefined ? true : options.removeValidationRecommendations
+            this.validateSchemaFunction = options.validateFunction || validate
+            this.rootValue = options.rootValue
+            this.contextFunction = options.contextFunction || this.defaultContextFuncion
+            this.fieldResolver = options.fieldResolver
+            this.typeResolver = options.typeResolver
+            this.executeFunction = options.executeFunction || execute
+            this.extensionFunction = options.extensionFunction || this.defaultExtensions
+            this.shouldUpdateSchemaFunction = options.shouldUpdateSchemaFunction || this.defaultShouldUpdateSchema
+            this.setSchema(options.schema)
+        }
     }
 
-    getSchema(): GraphQLSchema {
+    getSchema(): GraphQLSchema | undefined {
         return this.schema
     }
 
-    updateSchema(schema: GraphQLSchema): void {
-        this.schema = schema
+    isValidSchema(schema?: GraphQLSchema): boolean {
+        return schema ? this.schemaValidationErrors.length === 0 : false
+    }
+
+    setSchema(schema?: GraphQLSchema): void {
+        this.logger.info('Trying to set graphql schema')
+        this.logDebugIfEnabled(`Schema is  ${JSON.stringify(schema)}`)
+        if (this.shouldUpdateSchemaFunction(schema)) {
+            this.schema = schema
+            // Validate schema
+            if (this.schema) {
+                this.schemaValidationErrors = this.schemaValidationFunction(this.schema)
+                if (this.schemaValidationErrors.length > 0) {
+                    this.logger.warn('Schema validation failed with errors. Please check the GraphQL schema and fix potential issues.')
+                    for(const error of this.schemaValidationErrors) {
+                        this.logger.error('A schema validation error occurred: ', error)
+                        this.collectErrorMetricsFunction(SCHEMA_VALIDATION_ERROR, error, undefined)
+                    }
+                }
+            }
+        } else {
+            this.logger.warn('Schema update was rejected because condition set in "shouldUpdateSchema" check was not fulfilled.')
+        }
+
+        this.metricsClient.setAvailability(this.isValidSchema(this.schema) ? 1 : 0)
+    }
+
+    /** Defines whether a schema update should be executed. Default behaviour: If schema is undefined return false.
```

Comment:
I think the comment is a bit misleading, since the function does not _define_ (in the sense of setting) anything but only _determines_ whether the provided schema is defined or undefined.

Reply:
Changed